src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -66,20 +66,16 @@ default String getAsTextShowAll() {
      */
     default String getAsTextHidePrivate() {
         final StringBuilder builder = new StringBuilder();
-        builder.append(getName());
-        if (!getPhone().isPrivate()) {
-            builder.append(" Phone: ").append(getPhone());
-        }
-        if (!getEmail().isPrivate()) {
-            builder.append(" Email: ").append(getEmail());
-        }
-        if (!getAddress().isPrivate()) {
-            builder.append(" Address: ").append(getAddress());
-        }
+        builder.append(getPrintableString(getName(), getPhone(), getEmail(), getAddress()));
+        builder.append(" ");
         builder.append(" Tags: ");
         for (Tag tag : getTags()) {
             builder.append(tag);
         }
         return builder.toString();
     }
+
+    default String getPrintableString(Printable name, Printable phone, Printable email, Printable address) {
```

Comment:
This is not what is asked in the task. The method signature should be getPrintableString(Printable...). Take note of the varargs. You should have a loop in this method to handle the Printables.